code/__HELPERS/mob_bodyzone.dm

```diff
@@ -0,0 +1,142 @@
+
+/// Displays a UI around the target allowing the user to select which bodypart
+/// that they want to act on.
+/// Target: The location to show the user interface.
+/// Precise: Toggle to include groin, eyes and mouth. If true, implies hide_non_present will be forced to false.
+/// Icon Callback: The callback to run in order to get the selection zone overlay.
+/// If you want to wait for the result use: AWAIT(select_bodyzone(target))
+/mob/proc/select_bodyzone_from_wheel(atom/target, precise = FALSE, datum/callback/icon_callback, override_zones = null)
+	DECLARE_ASYNC
+	if (!client || !client.prefs)
+		ASYNC_RETURN(null)
+	if (!icon_callback)
+		icon_callback = CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(select_bodyzone_limb_default))
+	// Determine what parts we want to show
+	var/list/bodyzone_options = list()
+	var/list/parts = list(BODY_ZONE_HEAD, BODY_ZONE_L_ARM, BODY_ZONE_L_LEG, BODY_ZONE_CHEST, BODY_ZONE_R_LEG, BODY_ZONE_R_ARM)
+	if (override_zones)
+		parts = override_zones
+	for (var/bodyzone in parts)
+		var/image/created_image = image(icon = ui_style2icon(client.prefs?.read_player_preference(/datum/preference/choiced/ui_style)), icon_state = "zone_sel")
+		var/selection_overlay = icon_callback.Invoke(src, target, bodyzone, FALSE)
+		created_image.overlays += selection_overlay
+		bodyzone_options[bodyzone] = created_image
+	var/result = show_radial_menu(src, target, bodyzone_options, radius = 40, require_near = TRUE, tooltips = TRUE)
+
+	// Disconnected or no result
+	if (!result || !client)
+		ASYNC_RETURN(null)
+
+	// Let the user choose more zones
+	var/list/suboptions = null
+	if (precise && result == BODY_ZONE_HEAD)
+		suboptions = list(BODY_ZONE_HEAD, BODY_ZONE_PRECISE_EYES, BODY_ZONE_PRECISE_MOUTH)
+	else if (precise && result == BODY_ZONE_CHEST)
+		suboptions = list(BODY_ZONE_CHEST, BODY_ZONE_PRECISE_GROIN)
+
+	if (suboptions)
+		bodyzone_options = list()
+		for (var/bodyzone in suboptions)
+			var/image/created_image = image(icon = ui_style2icon(client.prefs?.read_player_preference(/datum/preference/choiced/ui_style)), icon_state = "zone_sel")
+			var/selection_overlay = icon_callback.Invoke(src, target, bodyzone, !(bodyzone in parts))
+			created_image.overlays += selection_overlay
+			bodyzone_options[bodyzone] = created_image
+		result = show_radial_menu(src, target, bodyzone_options, radius = 40, require_near = TRUE, tooltips = TRUE)
+
+	// Disconnected or no result
+	if (!result || !client)
+		ASYNC_RETURN(null)
+	if (!(result in parts))
+		ASYNC_RETURN(null)
+	ASYNC_RETURN(result)
+
+/proc/select_bodyzone_limb_default( mob/user, atom/target, bodyzone, is_precise_part = FALSE)
+	// Create the overlay
+	var/image/selection_overlay = image(icon = 'icons/mob/zone_sel.dmi', icon_state = bodyzone)
+	return selection_overlay
+
+/proc/select_bodyzone_limb_health(accurate_health = FALSE, mob/user, atom/target, bodyzone, is_precise_part = FALSE)
+	// Get the colours
+	var/list/healthy = rgb2num(FULL_HEALTH_COLOUR)
```

Comment:
If you define something only to immediately rgb2num it why not define it in the correct format

Reply:
Well the VS code plugin I have doesn't handle the byond specific list() format of colours compared to RGB colours, so it makes it a lot easier to see what colours have been defined.
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/8af3fdbe-588d-4a8b-94c4-239f9f64d93e)
Compared to
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/f927ecd4-4c6f-4ae4-8751-064e868fd1a2)

Reply:
Do you need it in list() format? BYOND supports hex colors

Reply:
I am generating a colour matrix from it because the target icon is red instead of white

Reply:
Change it to whiteee